core/blocktree/node.go

```diff
@@ -25,37 +25,37 @@ import (
 	"github.com/disiqueira/gotree"
 )
 
-// node is an element in the BlockTree
-type node struct {
+// Node is an element in the BlockTree
+type Node struct {
```

Comment:
I don't think this needs to be exported?

Reply:
Yea, was wondering the same thing. I'm not seeing a difference running lint with it as it was before.

Reply:
The error is:
`trie/node.go:197:27: exported func Decode returns unexported type github.com/ChainSafe/gossamer/trie.node, which can be annoying to use (golint)`

And it makes sense, there are few exported methods that are returning a private struct. We should either make the struct public or the public methods private. The easiest was to make it public, so let me know what is your opinion.

Reply:
we should make those functions on `node` unexported in that case, since we shouldn't be using them outside of the `trie` package

Reply:
right, sure!

Reply:
Well, I tried. But there are several fields with the same name, eg:
```
// Root returns the root of the trie
func (t *Trie) root() node {
	return t.root
}
```
and function decode for ex already exist unexported in the same package, so after renaming I cant proceed.
```
func decode(r io.Reader) (node, error) {
func decode(r io.Reader, prev node) error {
```

I understand the concern, the having the methods unexported will be a much harder refactor, are you sure about it ?